eng/Subsets.props

```diff
@@ -529,13 +528,17 @@
 
   <!-- Packs sets -->
 
+  <ItemGroup>
```

Comment:
The tricky ordering of the different build steps looks fragile.

I am wondering whether it would be better to use multiple build.sh invocations to bootstrap the FreeBSD cross build. The first invocation would skip building parts that depend on last-known-good packages for target arch, the second invocation would then build the rest and use the packages produced by the first invocation.

Reply:
Yup, it is fragile as in, when we are not building packs subset, it fails the build on platforms using PublishSingleFile. Pulling Microsoft.NETCore.App.Host.sfxproj in clr subset is also not an option because that depends on libs+host to build first.

There is an existing bug that on the systems where we use PublishAot for ILCompiler, it still looks for apphost (https://github.com/dotnet/runtime/pull/80154#issuecomment-2060649529, it was actually a regression in .net9 p1). That makes the refactoring those parts a degree more delicate.

Reply:
I've changed it so that ordering only affect cross-OS builds. In that, it doesn't build the incorrect ilc on FreeBSD if they are not building packs subset.

Reply:
I've changed it so that ordering only affect cross-OS builds. In that, it doesn't build the incorrect ilc on FreeBSD if packs subset is not built.

Reply:
@jkotas, the ILCompiler.csproj build will be now deferred until after packs are built, only for freebsd/illumos/haiku OS set which do not provide Microsoft SDK. This requires single invocation of `./build.sh`.

The current state of this PR ([351856c](https://github.com/dotnet/runtime/pull/105004/commits/351856c73283c6b451aec83c055e4decf8030931)) is such that if we still don't like Subsets.props changes, then we can revert Subsets.props (alone) and document the following steps for the devs of those platforms:

```sh
# invocation 1 - build everything without ILCompiler support
$ ./build.sh -os freebsd -p:NativeAotSupported=false

# invocation 2 - build clr.tools and packs subsets again without NativeAotSupported
#                condition so ILCompiler.csproj can use live-built apphost.
$ ./build.sh clr.tools+packs -os freebsd
```

Reply:
Ideally, the cross-arch and cross-OS bootstrapping builds would produce a full-featured bits that are equivalent (including how the tools get published) to the natively compiled packages. I think achieving it via multiple invocations of build.sh flow would be easier to maintain than the ordering of the individual build steps.

Reply:
@Thefrank, @sec, basically this is the new workflow for community platforms for cross-OS build (e.g. for freebsd on linux): https://github.com/dotnet/runtime/pull/105004#discussion_r1737107996 (two steps). I tested it with this workflow https://github.com/am11/CrossRepoCITesting/actions/runs/10845732989/workflow, and artifacts are https://github.com/am11/CrossRepoCITesting/releases/tag/freebsd_10845732989. The `ilc` executable in ` runtime.freebsd-x64.Microsoft.DotNet.ILCompiler.10.0.0-ci.nupkg` shows:

```sh
$ unzip ~/Downloads/runtime.freebsd-x64.Microsoft.DotNet.ILCompiler.10.0.0-ci.nupkg
$ find . -name ilc -exec file {} \; | fold
./tools/ilc: ELF 64-bit LSB pie executable, x86-64, version 1 (FreeBSD), dynamic
ally linked, interpreter /libexec/ld-elf.so.1, for FreeBSD 14.0 (1400097), FreeB
SD-style, BuildID[sha1]=235f8e6220cbf08ec8c1966a415f82f6ed072e4e, with debug_inf
o, not stripped
```

Reply:
> $ ./build.sh -os freebsd -p:NativeAotSupported=false
> $ ./build.sh clr.tools+packs -os freebsd

I think we need to have a clean separation of responsibilities between the two commands. Otherwise, I expect it will collide with where https://github.com/dotnet/runtime/pull/107772 is trying to go. A good split may be:

- The first command should be partial build that produces packages that the full build expects to come from LKG toolsets when there is one for target platform
- The second command should be a full build that uses the packages produced by the first command

We can define a new subset for the first command to make the command line UX simpler and stable over time. This subset would not be on by default. Building this subset can drop the packages in a new directory. The full build can automatically pick the LKG toolset from there if the directory exists.

Ideally, it would be even possible for the first command to be debug/checked build and the second command to be release build or vice versa, and it should work for non-cross builds too. 

@agocke Thoughts?